### PR TITLE
Revert "Revert" Add reconciliation for Collector v1beta1 CRD (#2703)"…

### PR DESCRIPTION
**Pull request by pavolloffay**
… (#2800)"

This reverts commit 57024fcb2f1cee4f665d0f7c603e6c3979a1cc09.

Resolves https://github.com/open-telemetry/opentelemetry-operator/issues/1907

Adds https://github.com/open-telemetry/opentelemetry-operator/pull/2703

**Description:** <Describe what has changed.>

**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #issue-number

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by swiatekm-sumo — APPROVED** [review]
We need to remember to merge https://github.com/open-telemetry/opentelemetry-operator/pull/2563 before the next release. @iblancasa can you rebase it once this PR is merged?

**Comment by jaronoff97** [comment]
We also want to merge this after this merges right? https://github.com/open-telemetry/opentelemetry-collector-releases/pull/507

**Pull request merged**
